rest/config_legacy_test.go

```diff
@@ -87,6 +117,104 @@ func TestLegacyConfigToStartupConfig(t *testing.T) {
 	}
 }
 
+func TestGocbv1tov2AddressUpgrade(t *testing.T) {
+	testCases := []struct {
+		name             string
+		server           string
+		expectError      bool
+		expectedServer   string
+		expectedUsername string
+		expectedPassword string
+	}{
+		{
+			name:             "Keep couchbase ports unless 8091",
+			server:           "couchbase://localhost:8091,localhosttwo:1234?network=true",
+			expectError:      false,
+			expectedServer:   "couchbase://localhost,localhosttwo:1234?network=true",
```

Comment:
We should be preserving _all_ custom ports (including `8091`) if the original scheme was `couchbase(s)://` - only stripping `8091` when we know it was `http://`

This is extremely unlikely to be hit (would require somebody to be running the memcached port on `8091`), but having this special case in here doesn't make sense.

```suggestion
			name:             "Keep couchbase ports",
			server:           "couchbase://localhost:8091,localhosttwo:1234?network=true",
			expectError:      false,
			expectedServer:   "couchbase://localhost:8091,localhosttwo:1234?network=true",
```